google-cloud-firestore/lib/google/cloud/firestore.rb

```diff
@@ -21,6 +21,442 @@ module Cloud
+    # end
+    # ```
+    #
+    # The example above is actually a simple query without filters. Let's look
+    # at some other queries for Cloud Firestore.
+    #
+    # ## Querying data
+    #
+    # Use {Firestore::Query#where} to filter queries on a field:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a collection reference
+    # cities_col = firestore.col "cities"
+    #
+    # # Create a query
+    # query = cities_col.where(:population, :>=, 1000000)
+    #
+    # query.get do |city|
+    #   puts "#{city.document_id} has #{city[:population]} residents."
+    # end
+    # ```
+    #
+    # You can order the query results with {Firestore::Query#order}:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a collection reference
+    # cities_col = firestore.col "cities"
+    #
+    # # Create a query
+    # query = cities_col.order(:name, :desc)
+    #
+    # query.get do |city|
+    #   puts "#{city.document_id} has #{city[:population]} residents."
+    # end
+    # ```
+    #
+    # Query methods may be chained, as in this example using
+    # {Firestore::Query#limit} and  {Firestore::Query#offset} to perform
+    # pagination:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a collection reference
+    # cities_col = firestore.col "cities"
+    #
+    # # Create a query
+    # query = cities_col.limit(5).offset(10)
+    #
+    # query.get do |city|
+    #   puts "#{city.document_id} has #{city[:population]} residents."
+    # end
+    # ```
+    #
+    # See [Managing Indexes in Cloud
+    # Firestore](https://cloud.google.com/firestore/docs/query-data/indexing) to
+    # ensure the best performance for your queries.
+    #
+    # ## Updating data
+    #
+    # You can use {Firestore::DocumentReference#set} to completely overwrite an
+    # existing document:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a document reference
+    # nyc_ref = firestore.doc "cities/NYC"
+    #
+    # nyc_ref.set({ name: "New York City" })
+    # ```
+    #
+    # Or, to selectively update only the fields appearing in your `data`
+    # argument, set the `merge` option to `true`:
+    #
+    # ```ruby
+    # require "google/cloud/firestore"
+    #
+    # firestore = Google::Cloud::Firestore.new
+    #
+    # # Get a document reference
+    # nyc_ref = firestore.doc "cities/NYC"
+    #
+    # nyc_ref.set({ name: "New York City" }, merge: true)
+    # ```
+    #
+    # Using {Firestore::DocumentReference#update} instead of `set` ensures that
+    # the provided data is merged into the existing document data. Unlike `set`,
+    # the `update` operation will fail if the document does not exist.
```

Comment:
The main differentiator is that update allows for direct deep-updates using field paths.

Reply:
Fixed.

Reply:
Text and example have been replaced with demonstration of `update` using `FieldPath`.